core/src/main/java/tc/oc/pgm/variables/types/PlayerLocationVariable.java

```diff
@@ -1,17 +1,35 @@
 package tc.oc.pgm.variables.types;
 
 import static java.lang.Math.toRadians;
+import static tc.oc.pgm.util.nms.PlayerUtils.PLAYER_UTILS;
 
+import java.util.HashMap;
 import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.block.RayBlockIntersection;
+import tc.oc.pgm.util.nms.NMSHacks;
 import tc.oc.pgm.variables.VariableDefinition;
 
 public class PlayerLocationVariable extends AbstractVariable<MatchPlayer> {
 
   private final Component component;
+  private final HashMap<MatchPlayer, RayBlockIntersection> locationTargetMap;
```

Comment:
i would not bother caching a map, just cache literally the latest intersection:
```
Location inputCache;
RayBlockIntersection resultCache;
```

then just:
```
private RayBlockIntersection(Player player) {
  if (Objects.equals(player, inputCache) return resultCache;
  inputCache = player.getLocation().clone();
  return resultCache = PLAYER_UTILS.getTargetedBlock(player);
}
```

Reply:
that would be easier